### PR TITLE
[SplashScreen] Add a new sample

### PR DESCRIPTION
**Pull request by yaraki**
Change-Id: Ieb43b88302a449778828da81e8bee64d4652d480

### CODE REVIEW

File: SplashScreen/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+dependencies {
+    implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+
+    implementation 'androidx.core:core-ktx:1.3.2'
+    implementation 'androidx.activity:activity-ktx:1.3.0-alpha05'
+    implementation 'androidx.appcompat:appcompat:1.3.0-beta01'
+    implementation 'com.google.android.material:material:1.3.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
```
Comment (chrisbanes) [inline SplashScreen/app/build.gradle:53]: Need constraintlayout?

File: SplashScreen/app/src/main/java/com/example/android/splashscreen/CustomActivity.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        // This callback is called when the app is ready to draw its content and replace the splash
+        // screen. We can customize the exit animation of the splash screen here.
+        splashScreen.setOnExitAnimationListener { splashScreenView ->
+
+            // The animated vector drawable is already animating at this point. Depending on the
+            // duration of the app launch, the animation might not have finished yet. We can
+            // calculate the remaining duration of the icon animation as follows.
+            val remainingDuration = (
+                    splashScreenView.iconAnimationDurationMillis -
```
Comment (chrisbanes) [inline SplashScreen/app/src/main/java/com/example/android/splashscreen/CustomActivity.kt:43]: This looks a bit weird to me. Can this be extracted to an ext-val?

File: SplashScreen/app/src/main/java/com/example/android/splashscreen/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    private val viewModel: MainViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val binding = MainActivityBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setSupportActionBar(binding.toolbar)
+        // Suspend drawing the app content until the initial data is ready.
+        suspendDraw()
```
Comment (chrisbanes) [inline SplashScreen/app/src/main/java/com/example/android/splashscreen/MainActivity.kt:30]: I'd call this `suppressDraw()` instead. `suspend` makes me think of coroutines.

File: SplashScreen/app/src/main/java/com/example/android/splashscreen/MainViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+     * True when the initial data in ViewModel is ready, and the app is ready to draw its main
+     * content.
+     */
+    var isReady = false
+        private set
+
+    init {
+        viewModelScope.launch {
+            val nightMode = withContext(Dispatchers.IO) {
+                prefs.getInt(PREF_NIGHT_MODE, UiModeManager.MODE_NIGHT_AUTO)
```
Comment (chrisbanes) [inline SplashScreen/app/src/main/java/com/example/android/splashscreen/MainViewModel.kt:58]: I think `getSharedPreferences()` should probably be done in the bg too

**Review by chrisbanes — COMMENTED** [review]
LGTM

File: SplashScreen/app/build.gradle (hunk 1 above)
Comment (yaraki) [reply SplashScreen/app/build.gradle]: No. Removed.

File: SplashScreen/app/src/main/java/com/example/android/splashscreen/CustomActivity.kt (hunk 2 above)
Comment (yaraki) [reply SplashScreen/app/src/main/java/com/example/android/splashscreen/CustomActivity.kt]: Extracted the calculation and added comments about each of the involved values.

File: SplashScreen/app/src/main/java/com/example/android/splashscreen/MainActivity.kt (hunk 3 above)
Comment (yaraki) [reply SplashScreen/app/src/main/java/com/example/android/splashscreen/MainActivity.kt]: Renamed.

File: SplashScreen/app/src/main/java/com/example/android/splashscreen/MainViewModel.kt (hunk 4 above)
Comment (yaraki) [reply SplashScreen/app/src/main/java/com/example/android/splashscreen/MainViewModel.kt:63]: Done. It is now in a background thread.

**Review by chrisbanes — APPROVED** [review]
LGTM. I assume that Vadim has already taken a look?

**Comment by yaraki** [comment]
Yes, it was 172330 internally.

**Pull request merged**